os/coreos-layering.md

```diff
@@ -0,0 +1,49 @@
+# CoreOS Layering
+
+This enhancement proposes:
+
+- A fundamental new addition to ostree/rpm-ostree, which is support
+  for directly pulling and updating the OS from container images (while keeping
+  all existing functionality, per-node package layering and
+  our existing support for pulling via ostree on the wire).
+- Documentation for generating derived (layered) images from the pristine CoreOS
+  base image.
+- Support for switching a FCOS system to use a custom image
+  on firstboot via Ignition
+- zincati will continue to perform upgrades by inspecting the upgrade graph
+  from the base image.
+
+# Existing work
+
+Originally, https://github.com/coreos/fedora-coreos-tracker/issues/812
+tracked native support for "encapsulating" ostree commits in containers.
+
+Then, we realized that when shipping the OS in a container image,
+it feels natural to support users deriving from it.
+
+The bulk of this is really "ostree native container" integration
+glue, and is happening in https://github.com/ostreedev/ostree-rs-ext
+
+rpm-ostree will vendor the ostree-rs-ext code and also be extended
+to support the same interfaces as implemented by the "base" ostree-rs-ext
+code.
+
+Specifically as of today, this functionality is exposed in:
+
+`rpm-ostree ex-container`
+`rpm-ostree rebase --experimental $containerref`
+
+# Rationale
+
+Since the creation of Container Linux (CoreOS) as well as Atomic Host, and
+continuing into the Fedora/RHEL CoreOS days, we have faced a constant
+tension around what we ship in the host system.
+
+[This issue](https://github.com/coreos/fedora-coreos-tracker/issues/401)
+encapsulates much prior discussion.
+
+For users who are happy with Fedora CoreOS today, not much will change.
+
+For those who e.g. want to install custom agents or nontrivial amounts
+of code (such as kubelet), this "CoreOS layering" will be a powerful
+new mechanism to ship the code they need.
```

Comment:
Might be worth adding an example Containerfile of what derivation could look like even if it's not how it'll look in the end. But at least it'd give a general idea.